finetrainers/trainer.py

```diff
@@ -314,7 +318,7 @@ def collate_fn(batch):
                 image_or_video=data["video"].unsqueeze(0),
                 device=self.state.accelerator.device,
                 dtype=self.state.weight_dtype,
-                generator=self.state.generator,
+                generator=torch.Generator(self.state.accelerator.device).manual_seed(self.args.seed),
```

Comment:
Our current `self.state.generator` is an advanced one which means once it's called the first time, it's intermediate states get changed and we cannot control how. It's better to initialize one every time we have to pass a generator, instead. 

This is more reproducible and explicit, IMO.

Reply:
It does not matter if the generator advances. If we start with the same seed and same starting conditions, a generator will always advance with the same internal states. Would revert to using `self.state.generator`.